models/llama4/moe.py

```diff
@@ -25,18 +25,24 @@ def __init__(
         num_local_experts: int,
         dim: int,
         hidden_dim: int,
+        int4_weight: bool = False,
     ) -> None:
         super().__init__()
 
+        self.int4_weight = int4_weight
         dtype = torch.get_default_dtype()
+        if int4_weight:
```

Comment:
this feels like complexity that truly doesn't belong at this layer. can we please keep it outside into quantization code somehow?

Reply:
we don't want `llama-models` to become `torchao` or `vllm` or whatever really. it is not a full fledged all powerful inference engine.